content/blog/2024/10/what-is-the-esp-registry/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "What is the ESP-Registry?"
+date: 2024-10-27
+showAuthor: false
+authors:
+  - "pedro-minatel"
+tags: ["Registry", "Component", "ESP-IDF"]
+---
+
+When starting a new project, we often reuse code from existing components, such as libraries and drivers, to save time. This not only saves time but also makes your project easier to maintain.
+
+In this article, we will explore what the ESP-Registry is and the key concepts of components, making it easy to understand not only how to use components but also how to contribute.
+
+## Concept
+
+In the [ESP-IDF](https://idf.espressif.com), components are widely used to make the framework more modular, making it clearer to understand and maintain. The naming convention in the ESP-IDF is to use the term "component" instead of "library" because the ESP-IDF is more like an operating system with self-contained modules that interact with the system at a higher level.
+
+There are three concepts that we need to introduce in order to understand the complete solution:
+
+### 1. Component
+
+As mentioned before, the component is a self-contained piece of code that can be used to add a new functionality to the application. In most of the cases, the component is used in several applications, making easy to reuse code and as well will improve the efficiency and reduce the development complexity.
+
+For example, in your application (`my-application`), you are going to use a specific sensor (`sensor-component`) that will collect temperature and humidity from the environment. Instead of adding all the logic behind the communication protocol, data collection, conversions and etc to the application code, you can add as a component.
+
+```text
+.
+└── my-application
+    ├── components
+    │   └── sensor-component
+    │       └── include
+    │           └── sensor-component.c
+    │       └── src
+    │           └── CMakeLists.txt
+    │           └── sensor-component.c
+    │       └── CMakeLists.txt
+    ├── CMakeLists.txt
+    ├── include
+    │   └── my-application.h
+    └── my-application.c
+```
+
+In this example, the `sensor-component` will be added to the project as a component, and can be shared as is among different projects that uses the same sensor.
+
+### 2. Component Manager
+
+To improve the developer experience, Espressif has developed a set of tools to help manage the component inside the project. The Component Manager is a CLI tool that allows you to create new components and add existing components to your project.
+
+#### Creating new components
+
+To create a new component using the component manager, you can use:

```

Comment:
As we discussed it, maybe it is worth dropping this subsection, because this information is not sufficient on its own. It is better to have it in a how to guide with more details about how to create a component.

Reply:
As an alternative, the command can be:

`idf.py create-component -C ./components component-name`

Reply:
Let's keep this to the next article.